bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitSettings.java

```diff
@@ -78,10 +82,21 @@ public void fill(Dictionary<String, ?> properties) throws UnknownHostException {
         if (maximumTemperature != null) {
             this.maximumTemperature = Double.parseDouble(maximumTemperature.toString());
         }
-        this.thermostatHeatMode = (String) properties.get("thermostatHeatMode");
-        this.thermostatCoolMode = (String) properties.get("thermostatCoolMode");
-        this.thermostatAutoMode = (String) properties.get("thermostatAutoMode");
-        this.thermostatOffMode = (String) properties.get("thermostatOffMode");
+        this.thermostatTargetModeHeat = Optional.ofNullable((String) properties.get("thermostatTargetModeHeat"))
+                .orElseGet(() -> (String) properties.get("thermostatHeatMode" /* legacy setting */));
+        this.thermostatTargetModeCool = Optional.ofNullable((String) properties.get("thermostatTargetModeCool"))
+                .orElseGet(() -> (String) properties.get("thermostatCoolMode" /* legacy setting */));
+        this.thermostatTargetModeAuto = Optional.ofNullable((String) properties.get("thermostatTargetModeAuto"))
+                .orElseGet(() -> (String) properties.get("thermostatAutoMode" /* legacy setting */));
+        this.thermostatTargetModeOff = Optional.ofNullable((String) properties.get("thermostatTargetModeOff"))
+                .orElseGet(() -> (String) properties.get("thermostatOffMode" /* legacy setting */));
+
+        Optional.ofNullable((String) properties.get("thermostatCurrentModeCooling"))
+                .ifPresent(value -> this.thermostatCurrentModeCooling = value);
```

Comment:
I'd just stick to doing if-then null checks instead using optionals for this. It's also done like that for the other fields.